### DISCUSSION TITLE
question about getSkyWalkingDynamicField, setSkyWalkingDynamicField

### DISCUSSION THREAD

**Discussion by abhinandc123** [apache/skywalking#12002] [Q&A]
hii i am reading src code of java agent, unable to understand how this works 
can anyone pls explain 
Link to class :- https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/EnhancedInstance.java

**Comment by CodePrometheus** [comment]
My understanding is that this is an extension field provided by skywalking for the target class, the implementation is based on bytebuddy `defineField`. It can be used to fill in span information and other purposes. You can look at the specific usage in the plugins implemented in java, which may help to understand.

**Reply by abhinandc123** [reply]
i was not able step into the interceptor also while debugging.

**Reply by CodePrometheus** [reply]
Debug by IDEA? Maybe this discussion https://github.com/apache/skywalking/discussions/11835 can help you

**Comment by abhinandc123** [comment]
`    @Override
    public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes,
        Object ret) {
        if (ret instanceof EnhancedInstance) {`
https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/mysql-8.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v8/ConnectionCreateInterceptor.java#L41

any idea where this object get convert to `EnhancedInstance`
